services/content/service.go

```diff
@@ -187,7 +187,7 @@ func (s *service) Read(req *api.ReadContentRequest, session api.Content_ReadServ
 
 	var (
 		offset = req.Offset
-		size   = req.Size_
+		size   = req.Size_ // this is buffer size, not the expected size of the blob!
```

Comment:
This is the read size.

Reply:
You can read this in https://github.com/containerd/containerd/blob/master/api/services/content/v1/content.proto#L143. It is the total size of the read.

Reply:
updated